app/controllers/agentAppointment/AppointAgentController.scala

```diff
@@ -306,40 +365,45 @@ class AppointAgentController @Inject()(representations: PropertyRepresentationCo
                                              organisationId: Long,
                                              individualId: Long,
                                              checkPermission: AgentPermission,
-                                             challengePermission: AgentPermission)(implicit hc: HeaderCarrier): Future[Unit] = {
-
-    propertyLinks.get(organisationId, pLink.toLong) map {
-      case Some(prop) => {
-        updateAllAgentsPermission(
-          pLink.toLong, prop,
-          AppointAgent(None, "", checkPermission, challengePermission),
-          agentOrgId,
-          individualId,
-          organisationId)
+                                             challengePermission: AgentPermission,
+                                             isAgent: Boolean)(implicit hc: HeaderCarrier): Future[Unit] = {
+    val link = if(isAgent) propertyLinks.getClientLink(pLink) else propertyLinks.getOwnerLink(pLink)
+
+    link map {
+        case Some(prop) => {
+          updateAllAgentsPermission(
+            pLink.toLong, prop,
+            AppointAgent(None, "", checkPermission, challengePermission),
+            agentOrgId,
+            individualId,
+            organisationId)
+        }
+        // shouldn't be possible for user to select a bad property link
```

Comment:
who wrote that comment?

Reply:
These comments were already there. Do you want me to remove them or is the assumption wrong?

Reply:
the assumption is fine, however that should be logged at warn level.

Reply:
Added logging